File: 1277. Count Square Submatrices with All Ones/dp.java
class Solution {
    public int countSquares(int[][] matrix) {
        int res = 0, H = matrix.length, W = matrix[0].length;
    
        for(int y = 0; y < H; y++){
            for(int x = 0; x < W; x++){
                if(matrix[y][x] == 1){
                    if(y == 0 || x == 0){
                        res++;
                        continue;
                    }
                    
                    matrix[y][x] = Math.min(
                        matrix[y-1][x-1],
                        Math.min(
                            matrix[y-1][x],
                            matrix[y][x-1]
                        )
                    ) + 1;

                    res += matrix[y][x];
                }
            }
        }
    
        return res;
    }
}

File: 1277. Count Square Submatrices with All Ones/dp.js
/**
 * @param {number[][]} matrix
 * @return {number}
 */
 var countSquares = function(matrix) {
  
    let res = 0;
    const H = matrix.length;
    const W = matrix[0].length
    
    for(let i = 0; i < H; i++){
        for(let j = 0; j < W; j++){
            if(matrix[i][j] === 1){
                if(i === 0 || j === 0) res++
                else{
                     matrix[i][j] = Math.min(
                        matrix[i][j-1],
                        matrix[i-1][j],
                        matrix[i-1][j-1]
                    ) + 1
                    
                    res += matrix[i][j]
                }
            }
        }
    }
    
    return res;
};